### PR TITLE
PR merge automation

### PR DESCRIPTION
**Pull request by neykov**
Makes merging Github PRs in the Apache repos easier and less error-prone. Adds more context to the merge commit message - PR title and body.

Create a Github personal token at https://github.com/settings/tokens for smoother experience - see how to use it in sources.

### CODE REVIEW

File: scripts/am [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+if [ -z "$PR" ]; then
+    echo "Usage: am <PR number>"
+    exit 1
+fi
+
+if ! git branch | grep -q "* master"; then
+    echo "Must be on master branch"
+    exit 1
+fi
```
Comment (geomacy) [inline scripts/am]: Maybe worth adding an `else` with 
```
git fetch --all
git pull --rebase
```
so as to automate that step too.

File: scripts/am [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+# <absolute path to brooklyn-dist>/scripts/am $@
+
+set -e
+
+PR=$1
+: ${PR_REMOTE:=upstream}
+: ${APACHE_REMOTE:=apache}
+: ${CLOSES_MESSAGE:="Closes #"}
+
+REMOTE_URL=$(git remote get-url ${APACHE_REMOTE})
```
Comment (geomacy) [inline scripts/am]: Perhaps worth a comment above, requires git 2.7 or more?

File: scripts/am (hunk 1 above)
Comment (geomacy) [inline scripts/am]: or, better, `git fetch ${PR_REMOTE}`

File: scripts/am (hunk 2 above)
Comment (neykov) [inline scripts/am]: Haven't realized this is a recent addition - have added a fall back.

File: scripts/am (hunk 1 above)
Comment (neykov) [inline scripts/am]: This just makes sure we are on `master` (vs say `0.10.0`). Updating to remote master happens here https://github.com/apache/brooklyn-dist/pull/74/files#diff-7c4ab39c149b092f17e43d7f46616063R85.

**Comment by richardcloudsoft** [comment]
Nice!

**Pull request merged**